### PR TITLE
Fix stream responses

### PR DESCRIPTION
**Pull request by alextartan**
fix for #25

### CODE REVIEW

**Comment by alextartan** [comment]
The build is failing on

> The command "if [[ $TEST_COVERAGE == 'true' ]]; then composer test-coverage ; else composer test ; fi" exited with 0.
> 0.16s$ if [[ $CS_CHECK == 'true' ]]; then composer cs-check ; fi
>  [Seld\JsonLint\ParsingException]
>  "./composer.json" does not contain valid JSON
>  Parse error on line 1:
>  Test!
>  ^
>  Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
> The command "if [[ $CS_CHECK == 'true' ]]; then composer cs-check ; fi" exited with 1.

I didn't change `composer.json`.

Any hints?

File: test/Psr7ResponseTest.php [hunk 1]
```diff
@@ -45,7 +49,31 @@ public function testResponseToZend($body, $status, $headers)
...
+            }
+        }
+    }
+
+    /**
+     * @dataProvider getResponseData
+     */
+    public function testResponseToZendFromRealStream($body, $status, $headers)
+    {
+        $stream = new Stream(__DIR__ . '/../composer.json', 'wb+');
```
Comment (ezimuel) [inline test/Psr7ResponseTest.php]: Please don't use the `composer.json` file, use instead a random file generated in a temporary dir using [sys_get_temp_dir()](http://php.net/manual/en/function.sys-get-temp-dir.php) or a fixed file stored in a `test/TestAsset/` folder.

File: test/Psr7ResponseTest.php [hunk 2]
```diff
@@ -61,7 +61,7 @@ public function testResponseToZend($body, $status, $headers)
      */
     public function testResponseToZendFromRealStream($body, $status, $headers)
     {
-        $stream = new Stream(__DIR__ . '/../composer.json', 'wb+');
+        $stream = new Stream(__DIR__ . '/../test.file', 'wb+');
```
Comment (ezimuel) [inline test/Psr7ResponseTest.php]: See my previous comment.

File: test/Psr7ResponseTest.php [hunk 3]
```diff
@@ -78,6 +78,8 @@ public function testResponseToZendFromRealStream($body, $status, $headers)
                 $this->assertContains($value, $zendHeaders[$type]);
             }
         }
+
+        unlink(__DIR__ . '/../test.file');
```
Comment (ezimuel) [inline test/Psr7ResponseTest.php]: See my previous comment. If you use a temporary file generated on the fly you can even omit this. If you use a fixed file in `test/TestAsset` you need to omit this unlink.

**Review by ezimuel — CHANGES_REQUESTED** [review]

File: test/Psr7ResponseTest.php (hunk 1 above)
Comment (alextartan) [reply test/Psr7ResponseTest.php]: good catch! updated to use `tempnam(sys_get_temp_dir(), 'Test')`

**Comment by alextartan** [comment]
@ezimuel I fixed the issue you pointed out. The test is now using a temporary file. Also dropped the `unlink` as it's not needed anymore

**Review by ezimuel — APPROVED** [review]

**Comment by ezimuel** [comment]
@alextartan thanks for your contribution!

File: src/Psr7Response.php [hunk 4]
```diff
@@ -10,35 +10,64 @@
...
-            $psr7Response->getProtocolVersion(),
-            $psr7Response->getStatusCode(),
-            $psr7Response->getReasonPhrase(),
-            self::psr7HeadersToString($psr7Response),
-            (string) $psr7Response->getBody()
-        );
-        return ZendResponse::fromString($response);
+        $uri = $psr7Response->getBody()->getMetadata('uri');
+
+        if ($uri === static::URI_TEMP) {
```
Comment (Xerkus) [inline src/Psr7Response.php:34]: This looks wrong. What about default `php://memory` stream?

File: src/Psr7Response.php [hunk 5]
```diff
@@ -10,35 +10,64 @@
...
+        foreach ($psr7Response->getHeaders() as $headerName => $headerValues) {
+            $zendHeaders->addHeader(new GenericHeader($headerName, implode('; ', $headerValues)));
+        }
+
+        // set the status
+        $response->setStatusCode($psr7Response->getStatusCode());
+        // set the headers
+        $response->setHeaders($zendHeaders);
+        // set the stream
+        $response->setStream(fopen($uri, 'rb'));
```
Comment (Xerkus) [inline src/Psr7Response.php:61]: This is a bug, it will lose `php://memory` stream content and any other non-file based streams.

**Pull request merged**